### PR TITLE
Add parsing for metric hierarchies

### LINKED ISSUE #38: Parse the output of the hierarchy extension
The XML response parsing in #27 currently does not parse the information provided in the XML body by the hierarchy extension. This should be provided to support caching strategies.

### PR DESCRIPTION
**Pull request by davidor**
Fixes #38

### CODE REVIEW

File: src/response.rs [hunk 1]
```diff
@@ -20,6 +21,19 @@ impl From<PeriodTime> for SystemTime {
     }
 }
 
+#[derive(Debug, PartialEq)]
+pub struct MetricsHierarchy(HashMap<String, Vec<String>>);
```
Comment (davidor) [inline src/response.rs]: I think we should have define the `Metric` type, but that affects other parts of the code, so I'd rather change that in a different PR.

File: src/response.rs [hunk 2]
```diff
@@ -134,6 +151,48 @@ impl<'de> Deserialize<'de> for PeriodTime {
...
+    fn visit_map<V>(self, mut map: V) -> Result<Self::Value, V::Error>
+    where
+        V: MapAccess<'de>,
+    {
+        let mut hierarchy = MetricsHierarchy::new();
+
+        // The key in the hierarchy structure is always "metric". It is not
+        // used, but we need to read it to get the value.
+        let mut key: Option<String> = map.next_key()?;
+        while key != None {
```
Comment (unleashed) [inline src/response.rs]: maybe `while map.next_key()?.is_some() {` or `while let Some(_) = map.next_key()? {` and you can remove the other two lines referencing `key`?

Comment (davidor) [reply src/response.rs]: Those do not work as they are. There are some missing type annotations.

Comment (davidor) [reply src/response.rs]: Fixed :+1:

**Comment by unleashed** [comment]
@davidor maybe add some helper to extract the data as well (maybe another PR?)

**Pull request merged**